## importing the library

In [1]:
#numpy -> To perform the mathematical operations
import numpy as np

#Padas -> Data Manipulattion tool
import pandas as pd

#Matplotlib -> Data Visualization tool
import matplotlib.pyplot as plt

#seaborn-> Data Visualization 
import seaborn as sns

#sqlite -> Server-less Database
import sqlite3

#Hints
1. CONNECT YOUR DATBASE (OR CREATE A CONNECTION )->sqlite3.connect(database)
2. use the curser function -> database _variable.curser()

#workflow.
1.you need to establish a connection to the database by creating a connewction object
2.then ,you have to crate a curser object using the curser() method
3.then ,excute the query -> curser_object.execute('query')
4.to fetch the data ,then use fetchall() method of the curser object 

# 1.Establishing the connection to the data base

In [2]:
db="movies.sqlite"
# Connect to SQLite database
connect = sqlite3.connect(db)
curs=connect.cursor()

a cursor object is used to interact with the database. The cursor acts as a mediator between the SQLite database connection and your Python code. It allows you to execute SQL queries, fetch data, and manage transactions.

2.Get all the data about movies

In [3]:
#Establishing the connection with the DatBase 
curs.execute("select * from movies")
# Retrieves all remaining rows of the query result. 
Imdb_movies=curs.fetchall()


In [12]:
#Displaying the first row content
Imdb_movies[0:5]

[(43597,
  'Avatar',
  237000000,
  150,
  '2009-12-10',
  2787965087,
  'Avatar',
  7.2,
  11800,
  'In the 22nd century, a paraplegic Marine is dispatched to the moon Pandora on a unique mission, but becomes torn between following orders and protecting an alien civilization.',
  'Enter the World of Pandora.',
  19995,
  4762),
 (43598,
  "Pirates of the Caribbean: At World's End",
  300000000,
  139,
  '2007-05-19',
  961000000,
  "Pirates of the Caribbean: At World's End",
  6.9,
  4500,
  'Captain Barbossa, long believed to be dead, has come back to life and is headed to the edge of the Earth with Will Turner and Elizabeth Swann. But nothing is quite as it seems.',
  'At the end of the world, the adventure begins.',
  285,
  4763),
 (43599,
  'Spectre',
  245000000,
  107,
  '2015-10-26',
  880674609,
  'Spectre',
  6.3,
  4466,
  'A cryptic message from Bond’s past sends him on a trail to uncover a sinister organization. While M battles political forces to keep the secret service 

In [7]:
#creating the dataframe 
df = pd.DataFrame(Imdb_movies, columns = ['id', 'original_title', 'budget', 'popularity', 'release_date',
'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline',
'uid', 'director_id' ])

In [8]:
#showing few datapoint
df.head(10)

id                            original_title     budget  popularity  \
0  43597                                    Avatar  237000000         150   
1  43598  Pirates of the Caribbean: At World's End  300000000         139   
2  43599                                   Spectre  245000000         107   
3  43600                     The Dark Knight Rises  250000000         112   
4  43601                               John Carter  260000000          43   
5  43602                              Spider-Man 3  258000000         115   
6  43603                                   Tangled  260000000          48   
7  43604                   Avengers: Age of Ultron  280000000         134   
8  43605    Harry Potter and the Half-Blood Prince  250000000          98   
9  43606        Batman v Superman: Dawn of Justice  250000000         155   

  release_date     revenue                                     title  \
0   2009-12-10  2787965087                                    Avatar   
1   2007-05-19   961000000  Pirates of the Caribbean: At World's End   
2   2015-10-26   880674609                                   Spectre   
3   2012-07-16  1084939099                     The Dark Knight Rises   
4   2012-03-07   284139100                               John Carter   
5   2007-05-01   890871626                              Spider-Man 3   
6   2010-11-24   591794936                                   Tangled   
7   2015-04-22  1405403694                   Avengers: Age of Ultron   
8   2009-07-07   933959197    Harry Potter and the Half-Blood Prince   
9   2016-03-23   873260194        Batman v Superman: Dawn of Justice   

   vote_average  vote_count  \
0           7.2       11800   
1           6.9        4500   
2           6.3        4466   
3           7.6        9106   
4           6.1        2124   
5           5.9        3576   
6           7.4        3330   
7           7.3        6767   
8           7.4        5293   
9           5.7        7004   

                                            overview  \
0  In the 22nd century, a paraplegic Marine is di...   
1  Captain Barbossa, long believed to be dead, ha...   
2  A cryptic message from Bond’s past sends him o...   
3  Following the death of District Attorney Harve...   
4  John Carter is a war-weary, former military ca...   
5  The seemingly invincible Spider-Man goes up ag...   
6  When the kingdom's most wanted-and most charmi...   
7  When Tony Stark tries to jumpstart a dormant p...   
8  As Harry begins his sixth year at Hogwarts, he...   
9  Fearing the actions of a god-like Super Hero l...   

                                          tagline     uid  director_id  
0                     Enter the World of Pandora.   19995         4762  
1  At the end of the world, the adventure begins.     285         4763  
2                           A Plan No One Escapes  206647         4764  
3                                 The Legend Ends   49026         4765  
4            Lost in our world, found in another.   49529         4766  
5                              The battle within.     559         4767  
6        They're taking adventure to new lengths.   38757         4768  
7                             A New Age Has Come.   99861         4769  
8                           Dark Secrets Revealed     767         4770  
9                              Justice or revenge  209112         4771

In [9]:
#check information non-null count and dtype 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 4773 entries, 0 to 4772
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   id              4773 non-null   int64  
 1   original_title  4773 non-null   object 
 2   budget          4773 non-null   int64  
 3   popularity      4773 non-null   int64  
 4   release_date    4773 non-null   object 
 5   revenue         4773 non-null   int64  
 6   title           4773 non-null   object 
 7   vote_average    4773 non-null   float64
 8   vote_count      4773 non-null   int64  
 9   overview        4770 non-null   object 
 10  tagline         3951 non-null   object 
 11  uid             4773 non-null   int64  
 12  director_id     4773 non-null   int64  
dtypes: float64(1), int64(7), object(5)
memory usage: 484.9+ KB


3.Get all the data about director 

In [13]:
#Establishing the connection with the DataBase 
curs.execute("select * from directors")
# create the cursor object 
directors=curs.fetchall()

In [14]:
#displaying the 10th row of director table
directors[0:10]

[('James Cameron', 4762, 2, 2710, 'Directing'),
 ('Gore Verbinski', 4763, 2, 1704, 'Directing'),
 ('Sam Mendes', 4764, 2, 39, 'Directing'),
 ('Christopher Nolan', 4765, 2, 525, 'Directing'),
 ('Andrew Stanton', 4766, 2, 7, 'Directing'),
 ('Sam Raimi', 4767, 2, 7623, 'Directing'),
 ('Byron Howard', 4768, 2, 76595, 'Directing'),
 ('Joss Whedon', 4769, 2, 12891, 'Directing'),
 ('David Yates', 4770, 2, 11343, 'Directing'),
 ('Zack Snyder', 4771, 2, 15217, 'Directing')]

In [15]:
#create the dataframe for director
director=pd.DataFrame(directors,columns=["name",'id','gender','uid','department'])

In [16]:
#showing 10th datapoint of director DataFrame
director.head(10)

name    id  gender    uid department
0      James Cameron  4762       2   2710  Directing
1     Gore Verbinski  4763       2   1704  Directing
2         Sam Mendes  4764       2     39  Directing
3  Christopher Nolan  4765       2    525  Directing
4     Andrew Stanton  4766       2      7  Directing
5          Sam Raimi  4767       2   7623  Directing
6       Byron Howard  4768       2  76595  Directing
7        Joss Whedon  4769       2  12891  Directing
8        David Yates  4770       2  11343  Directing
9        Zack Snyder  4771       2  15217  Directing

In [13]:
#check information non-null count and dtype 
director.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2349 entries, 0 to 2348
Data columns (total 5 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   name        2349 non-null   object
 1   id          2349 non-null   int64 
 2   gender      2349 non-null   int64 
 3   uid         2349 non-null   int64 
 4   department  2349 non-null   object
dtypes: int64(3), object(2)
memory usage: 91.9+ KB


4.Check how many Movies are present in the ImDb  table 

In [14]:
#Establishing the connection with the DataBase 
curs.execute("select Count(title) from movies")
#fetch the no. of count of movie
count=curs.fetchall()
print("The numbers of movies present in the iMDb dataset is",count[0])

The numbers of movies present in the iMDb dataset is (4773,)


5.find thses 3 director :john woo,James Cameron, Luc Besson

In [18]:
curs.execute("select * from directors WHERE name IN('Luc Besson','John Woo','James Cameron')")
#creating curser object 
Particular_director=curs.fetchall()
print(f'These 3 directors data are: {Particular_director}')

These 3 directors data are: [('James Cameron', 4762, 2, 2710, 'Directing'), ('John Woo', 4893, 2, 11401, 'Directing'), ('Luc Besson', 4949, 2, 59, 'Directing')]


6. Find all the directors with name starting with 'Steven'

In [19]:
curs.execute("SELECT * FROM directors WHERE name LIKE 'Steven%'")
Steven_director=curs.fetchall()
print(f"The directors whose names are starting with name 'Steven' are: {Steven_director}")

The directors whose names are starting with name 'Steven' are: [('Steven Spielberg', 4799, 2, 488, 'Directing'), ('Steven Soderbergh', 4909, 2, 1884, 'Directing'), ('Steven Brill', 5013, 2, 32593, 'Directing'), ('Steven Zaillian', 5117, 2, 2260, 'Directing'), ('Steven Quale', 5216, 2, 93214, 'Directing'), ('Steven Seagal', 5221, 2, 23880, 'Directing'), ('Steven E. de Souza', 5390, 2, 1726, 'Directing'), ('Steven Shainberg', 5803, 2, 67795, 'Directing'), ('Steven R. Monroe', 6713, 2, 88039, 'Directing')]


7.count the Female Directors

In [22]:
curs.execute("SELECT count(*) FROM directors WHERE gender=1")
female_director=curs.fetchall()
print(f"The No. of female director are: {female_director[0][0]}")

The No. of female director are: 150


8.Findthe name of the 10th first women director from table  

In [25]:
curs.execute("SELECT name FROM directors WHERE gender=1 LIMIT 10")
firstten_director=curs.fetchall()
print(f"The First top ten director are: {firstten_director[9][0]}")

The First top ten director are: Angelina Jolie


In [27]:
# ten female director
firstten_director

[('Brenda Chapman',),
 ('Lilly Wachowski',),
 ('Jennifer Yuh Nelson',),
 ('Kathryn Bigelow',),
 ('Nancy Meyers',),
 ('Jill Culton',),
 ('Mimi Leder',),
 ('Vicky Jenson',),
 ('Betty Thomas',),
 ('Angelina Jolie',)]

9. What are the 3 most popular movies ?

In [33]:
curs.execute("SELECT title FROM movies ORDER BY popularity DESC LIMIT 3")
threemovies=curs.fetchall()
print(f"The Most popular three movies are: {threemovies}")

The Most popular three movies are: [('Minions',), ('Interstellar',), ('Deadpool',)]


In [34]:
curs.execute("SELECT title FROM movies ORDER BY popularity DESC LIMIT 3")
threemovies=curs.fetchall()
print(f"The Most popular three movies are: {threemovies[0][0]} , {threemovies[1][0]} and {threemovies[2][0]}")

The Most popular three movies are: Minions , Interstellar and Deadpool


10. what are the 3 most bankable movies

In [35]:
curs.execute("SELECT title FROM movies ORDER BY budget DESC LIMIT 3")
threemovies=curs.fetchall()
print(f"The Most popular three movies are: {threemovies[0][0]} , {threemovies[1][0]} and {threemovies[2][0]}")

The Most popular three movies are: Pirates of the Caribbean: On Stranger Tides , Pirates of the Caribbean: At World's End and Avengers: Age of Ultron


In [36]:
threemovies

[('Pirates of the Caribbean: On Stranger Tides',),
 ("Pirates of the Caribbean: At World's End",),
 ('Avengers: Age of Ultron',)]

11. what is most 3 awarded average vote movie since the jan 1st ,2000

In [39]:
curs.execute("SELECT title FROM movies WHERE  release_date>='2000-01-01' ORDER BY vote_average DESC LIMIT 3")
mostawarded=curs.fetchall()
print(f"The Most awarded three movies are: {mostawarded[0][0]} , {mostawarded[1][0]} and {mostawarded[2][0]}")

The Most awarded three movies are: Sardaarji , Spirited Away and Whiplash


In [41]:
mostawarded

[('Sardaarji',), ('Spirited Away',), ('Whiplash',)]

12.which movies were directed by Brenda Chapman?

In [48]:
curs.execute("SELECT movies.original_title FROM movies JOIN directors ON movies.director_id=directors.id WHERE directors.name='Brenda Chapman'")
Directedmovies=curs.fetchall()
print(f"The movies which is directed by Brenda Chapman are: {Directedmovies[0][0]}")

The movies which is directed by Brenda Chapman are: Brave


In [46]:
Directedmovies

[('Brave',)]

13. Name the director  who has made the most movies

In [51]:
curs.execute("SELECT directors.name FROM movies JOIN directors on movies.director_id=directors.id group by Director_id order by count(name) DESC LIMIT 1")
Director_name=curs.fetchall()
print(f"Name the director who has made the most movies is: {Director_name[0][0]}")

Name the director who has made the most movies is: Steven Spielberg


In [52]:
df.columns

Index(['id', 'original_title', 'budget', 'popularity', 'release_date',
       'revenue', 'title', 'avg_ratings', 'rating_count', 'overview',
       'tagline', 'uid', 'director_id'],
      dtype='object')

14. Name of the director who is most bankable

In [56]:
curs.execute("SELECT directors.name FROM movies JOIN directors ON movies.director_id=directors.id group by Director_id order by SUM(budget) DESC LIMIT 1")
bankablemovies=curs.fetchall()
print(f"Name of the director who is most bankable is: {bankablemovies[0][0]}")

Name of the director who is most bankable is: Steven Spielberg


# Budget Analysis

1. tell the top 10 highest budget making movies

In [61]:
top_10=pd.DataFrame(curs.execute("SELECT * FROM movies ORDER BY budget DESC LIMIT 10"),columns=['id', 'original_title', 'budget', 'popularity', 'release_date',
'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline',
'uid', 'director_id' ])

In [62]:
top_10

id                               original_title     budget  popularity  \
0  43614  Pirates of the Caribbean: On Stranger Tides  380000000         135   
1  43598     Pirates of the Caribbean: At World's End  300000000         139   
2  43604                      Avengers: Age of Ultron  280000000         134   
3  43607                             Superman Returns  270000000          57   
4  43601                                  John Carter  260000000          43   
5  43603                                      Tangled  260000000          48   
6  43602                                 Spider-Man 3  258000000         115   
7  43610                              The Lone Ranger  255000000          49   
8  43600                        The Dark Knight Rises  250000000         112   
9  43605       Harry Potter and the Half-Blood Prince  250000000          98   

  release_date     revenue                                        title  \
0   2011-05-14  1045713802  Pirates of the Caribbean: On Stranger Tides   
1   2007-05-19   961000000     Pirates of the Caribbean: At World's End   
2   2015-04-22  1405403694                      Avengers: Age of Ultron   
3   2006-06-28   391081192                             Superman Returns   
4   2012-03-07   284139100                                  John Carter   
5   2010-11-24   591794936                                      Tangled   
6   2007-05-01   890871626                                 Spider-Man 3   
7   2013-07-03    89289910                              The Lone Ranger   
8   2012-07-16  1084939099                        The Dark Knight Rises   
9   2009-07-07   933959197       Harry Potter and the Half-Blood Prince   

   vote_average  vote_count  \
0           6.4        4948   
1           6.9        4500   
2           7.3        6767   
3           5.4        1400   
4           6.1        2124   
5           7.4        3330   
6           5.9        3576   
7           5.9        2311   
8           7.6        9106   
9           7.4        5293   

                                            overview  \
0  Captain Jack Sparrow crosses paths with a woma...   
1  Captain Barbossa, long believed to be dead, ha...   
2  When Tony Stark tries to jumpstart a dormant p...   
3  Superman returns to discover his 5-year absenc...   
4  John Carter is a war-weary, former military ca...   
5  When the kingdom's most wanted-and most charmi...   
6  The seemingly invincible Spider-Man goes up ag...   
7  The Texas Rangers chase down a gang of outlaws...   
8  Following the death of District Attorney Harve...   
9  As Harry begins his sixth year at Hogwarts, he...   

                                          tagline    uid  director_id  
0                     Live Forever Or Die Trying.   1865         4775  
1  At the end of the world, the adventure begins.    285         4763  
2                             A New Age Has Come.  99861         4769  
3                                            None   1452         4772  
4            Lost in our world, found in another.  49529         4766  
5        They're taking adventure to new lengths.  38757         4768  
6                              The battle within.    559         4767  
7                         Never Take Off the Mask  57201         4763  
8                                 The Legend Ends  49026         4765  
9                           Dark Secrets Revealed    767         4770

### Revenue Analysis

1. Find top 10 Revenue making movies

In [63]:
Top_revenue_movies=pd.DataFrame(curs.execute("SELECT * FROM movies ORDER BY revenue DESC LIMIT 10"),columns=['id', 'original_title', 'budget', 'popularity', 'release_date',
'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline',
'uid', 'director_id'])

In [64]:
Top_revenue_movies

id              original_title     budget  popularity release_date  \
0  43597                      Avatar  237000000         150   2009-12-10   
1  43622                     Titanic  200000000         100   1997-11-18   
2  43613                The Avengers  220000000         144   2012-04-25   
3  43625              Jurassic World  150000000         418   2015-06-09   
4  43641                   Furious 7  190000000         102   2015-04-01   
5  43604     Avengers: Age of Ultron  280000000         134   2015-04-22   
6  43721                      Frozen  150000000         165   2013-11-27   
7  43628                  Iron Man 3  200000000          77   2013-04-18   
8  44143                     Minions   74000000         875   2015-06-17   
9  43623  Captain America: Civil War  250000000         198   2016-04-27   

      revenue                       title  vote_average  vote_count  \
0  2787965087                      Avatar           7.2       11800   
1  1845034188                     Titanic           7.5        7562   
2  1519557910                The Avengers           7.4       11776   
3  1513528810              Jurassic World           6.5        8662   
4  1506249360                   Furious 7           7.3        4176   
5  1405403694     Avengers: Age of Ultron           7.3        6767   
6  1274219009                      Frozen           7.3        5295   
7  1215439994                  Iron Man 3           6.8        8806   
8  1156730962                     Minions           6.4        4571   
9  1153304495  Captain America: Civil War           7.1        7241   

                                            overview  \
0  In the 22nd century, a paraplegic Marine is di...   
1  84 years later, a 101-year-old woman named Ros...   
2  When an unexpected enemy emerges and threatens...   
3  Twenty-two years after the events of Jurassic ...   
4  Deckard Shaw seeks revenge against Dominic Tor...   
5  When Tony Stark tries to jumpstart a dormant p...   
6  Young princess Anna of Arendelle dreams about ...   
7  When Tony Stark's world is torn apart by a for...   
8  Minions Stuart, Kevin and Bob are recruited by...   
9  Following the events of Age of Ultron, the col...   

                                             tagline     uid  director_id  
0                        Enter the World of Pandora.   19995         4762  
1          Nothing on Earth could come between them.     597         4762  
2                            Some assembly required.   24428         4769  
3                                  The park is open.  135397         4783  
4                                Vengeance Hits Home  168259         4794  
5                                A New Age Has Come.   99861         4769  
6  Only the act of true love will thaw a frozen h...  109445         4844  
7                Unleash the power behind the armor.   68721         4784  
8       Before Gru, they had a history of bad bosses  211672         5045  
9                                    Divided We Fall  271110         4781

# voting analysis

1.Find the most popular movies with highest voting 

In [65]:
Top_rating_movies=pd.DataFrame(curs.execute("SELECT * FROM movies ORDER BY vote_average DESC LIMIT 10"),columns=['id', 'original_title', 'budget', 'popularity', 'release_date',
'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline',
'uid', 'director_id' ])

In [66]:
Top_rating_movies

id            original_title    budget  popularity release_date  \
0  47116          Stiff Upper Lips         0           0   1998-06-12   
1  47642     Dancer, Texas Pop. 81         0           0   1998-05-01   
2  47589                 Sardaarji         0           0   2015-06-26   
3  45983            One Man's Hero         0           0   1999-08-02   
4  45478  The Shawshank Redemption  25000000         136   1994-09-23   
5  46567        There Goes My Baby  10500000           0   1994-09-02   
6  46393     The Prisoner of Zenda         0           4   1937-09-03   
7  46934             The Godfather   6000000         143   1972-03-14   
8  44259                Fight Club  63000000         146   1999-10-15   
9  45415          Schindler's List  22000000         104   1993-11-29   

     revenue                     title  vote_average  vote_count  \
0          0          Stiff Upper Lips          10.0           1   
1     565592     Dancer, Texas Pop. 81          10.0           1   
2          0                 Sardaarji           9.5           2   
3          0            One Man's Hero           9.3           2   
4   28341469  The Shawshank Redemption           8.5        8205   
5     123509        There Goes My Baby           8.5           2   
6          0     The Prisoner of Zenda           8.4          11   
7  245066411             The Godfather           8.4        5893   
8  100853753                Fight Club           8.3        9413   
9  321365567          Schindler's List           8.3        4329   

                                            overview  \
0  Stiff Upper Lips is a broad parody of British ...   
1  Four guys, best friends, have grown up togethe...   
2  A ghost hunter uses bottles to capture trouble...   
3  One Man's Hero tells the little-known story of...   
4  Framed in the 1940s for the double murder of h...   
5  A group of high school seniors meets in the su...   
6  An Englishman on a Ruritarian holiday must imp...   
7  Spanning the years 1945 to 1955, a chronicle o...   
8  A ticking-time-bomb insomniac and a slippery s...   
9  The true story of how businessman Oskar Schind...   

                                             tagline     uid  director_id  
0                                               None   89861         6332  
1       in the middle of nowhere they had everything   78373         5463  
2                                               None  346081         6588  
3           One man's hero is another man's traitor.   69848         5770  
4  Fear can hold you prisoner. Hope can set you f...     278         5051  
5                                               None   88641         6051  
6       The most thrilling swordfight ever filmed...   43867         5972  
7                         An offer you can't refuse.     238         5179  
8                            Mischief. Mayhem. Soap.     550         4829  
9    Whoever saves one life, saves the world entire.     424         4799

2.Find all the movies for vote_average and vote_count

In [ ]:
Top_rating_movies=pd.DataFrame(curs.execute("SELECT * FROM movies ORDER BY vote_average and vote_count DESC LIMIT 10"),columns=['id', 'original_title', 'budget', 'popularity', 'release_date',
'revenue', 'title', 'vote_average', 'vote_count', 'overview', 'tagline',
'uid', 'director_id'])

# DIRECTOR ANALYSIS

1. Name all the directors with the numbers of movies and revenue where revenue should be taken into account forr doing the analysis. The director who has the highest revenue should comes at the top and so and so for

In [74]:
Top_directors=pd.DataFrame(curs.execute("SELECT directors.name,COUNT(*) as 'No. Of Movies',SUM(revenue) FROM movies JOIN directors ON movies.director_id=directors.id GROUP BY name ORDER BY sum(revenue) DESC"),columns=['Director','Total Movies',"Revenue"
])

In [77]:
Top_directors.head(10)

Director  Total Movies     Revenue
0   Steven Spielberg            27  9147393164
1      Peter Jackson             9  6498642820
2      James Cameron             7  5883569439
3        Michael Bay            12  5832524638
4  Christopher Nolan             8  4227483234
5     Chris Columbus            11  3725631503
6    Robert Zemeckis            13  3590622002
7       George Lucas             5  3339113893
8         Tim Burton            14  3337418241
9       Ridley Scott            16  3189557997

2.Name all the director with the number of movies and revenue where number of movies should be taken into account for doing the analysis.The director who has the number of movies should comes at the top and so and so for

In [78]:
Top_directors=pd.DataFrame(curs.execute("SELECT directors.name,COUNT(*) as 'No. Of Movies',SUM(revenue) FROM movies JOIN directors ON movies.director_id=directors.id GROUP BY name ORDER BY COUNT(*) DESC"),columns=['Director','Total Movies',"Revenue"
])

In [80]:
Top_directors.head(10)

Director  Total Movies     Revenue
0   Steven Spielberg            27  9147393164
1        Woody Allen            21   669101038
2    Martin Scorsese            20  1956635998
3     Clint Eastwood            20  2512058888
4          Spike Lee            16   340618771
5   Robert Rodriguez            16  1109899581
6       Ridley Scott            16  3189557997
7  Steven Soderbergh            15  2114864443
8       Renny Harlin            15  1031500590
9         Tim Burton            14  3337418241

3.Give the title of the movies, release_date,budget,revenue,popularity and vote_average made by Steven Spielberg

In [84]:
director_movies=pd.DataFrame(curs.execute("SELECT title,release_date,budget,revenue,popularity,vote_average FROM movies JOIN directors ON movies.director_id=directors.id where name Like 'Steven Spielberg'"),columns=['Title','Release_date','Budget','Revenue','popularity','vote_average'])

In [85]:
director_movies

Title Release_date     Budget  \
0   Indiana Jones and the Kingdom of the Crystal S...   2008-05-21  185000000   
1                                             The BFG   2016-06-01  140000000   
2                                   War of the Worlds   2005-06-28  132000000   
3                            The Adventures of Tintin   2011-10-25  130000000   
4                                     Minority Report   2002-06-20  102000000   
5                        A.I. Artificial Intelligence   2001-06-29  100000000   
6                       The Lost World: Jurassic Park   1997-05-23   73000000   
7                                        The Terminal   2004-06-17   60000000   
8                                              Munich   2005-12-22   70000000   
9                                                Hook   1991-12-11   70000000   
10                                          War Horse   2011-12-25   66000000   
11                                Saving Private Ryan   1998-07-24   70000000   
12                                            Lincoln   2012-11-09   65000000   
13                                      Jurassic Park   1993-06-11   63000000   
14                                Catch Me If You Can   2002-12-25   52000000   
15                 Indiana Jones and the Last Crusade   1989-05-24   48000000   
16                                    Bridge of Spies   2015-10-15   40000000   
17                                            Amistad   1997-12-03   36000000   
18                                               1941   1979-12-13   35000000   
19               Indiana Jones and the Temple of Doom   1984-05-23   28000000   
20                                   Schindler's List   1993-11-29   22000000   
21                            Raiders of the Lost Ark   1981-06-12   18000000   
22                 Close Encounters of the Third Kind   1977-11-16   20000000   
23                                   The Color Purple   1985-12-18   15000000   
24                                               Jaws   1975-06-18    7000000   
25                         E.T. the Extra-Terrestrial   1982-04-03   10500000   
26                           Twilight Zone: The Movie   1983-06-24   10000000   

      Revenue  popularity  vote_average  
0   786636033          75           5.7  
1   183345589          44           6.0  
2   591739379          48           6.2  
3   371940071          89           6.7  
4   358372926          65           7.1  
5   235926552          34           6.8  
6   229074524           2           6.2  
7   219417255          57           7.0  
8   130358911          29           6.9  
9   300854823          33           6.6  
10  177584879          29           7.0  
11  481840909          76           7.9  
12  275293450          36           6.7  
13  920100000          40           7.6  
14  352114312          73           7.7  
15  474171806          80           7.6  
16  165478348          48           7.2  
17   74000000           3           6.8  
18   31755742          10           5.6  
19  333000000          66           7.1  
20  321365567         104           8.3  
21  389925971          68           7.7  
22  303788635          52           7.2  
23  146292009          17           7.7  
24  470654000          50           7.5  
25  792910554          56           7.3  
26   29450919          12           6.2

### Conculusion

The Director 'Steven Spielberg' has done 27 number of movies and he generated 9147393164 total revenue.